### PR TITLE
FEAT: Allow cancel inactive limit orders

### PR DESCRIPTION
**Pull request by jcompagni10**
Allow inactive limit orders to be canceled instead of withdrawn. 
This new behavior has 2 benefits. 

1. Right now it takes at least 1 extra query to figure out if a tranche is inactive. If it's inactive you can currently only call withdraw. If we allow cancel you can always safely call cancel to completely remove your position.
2. This is very nice for astroport. Their goal is to get the value of a position at any given time. If they can always call SimulateCancel they can reduce this down to a single query instead of 2.

**related PRs**
https://github.com/neutron-org/neutron-integration-tests/pull/355

### CODE REVIEW

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/10728931355

**Comment by joldie777** [comment]
Please attach a succesfull testrun

File: x/dex/keeper/cancel_limit_order.go [hunk 1]
```diff
@@ -59,20 +59,20 @@ func (k Keeper) ExecuteCancelLimitOrder(
 ) (makerCoinOut, takerCoinOut sdk.Coin, tradePairID *types.TradePairID, error error) {
 	trancheUser, found := k.GetLimitOrderTrancheUser(ctx, callerAddr.String(), trancheKey)
 	if !found {
-		return sdk.Coin{}, sdk.Coin{}, nil, types.ErrActiveLimitOrderNotFound
+		return sdk.Coin{}, sdk.Coin{}, nil, types.ErrValidLimitOrderTrancheNotFound
```
Comment (sotnikov-s) [inline x/dex/keeper/cancel_limit_order.go]: according to the [error definition](https://github.com/neutron-org/neutron/blob/bb920dadbfa9a94bd7b4bf95ef86becc590b43ed/x/dex/types/errors.go#L27-L31), it's a wrapped error supposed to be here, isn't it? Here and at the line 75. Just like below at the line 88

**Review by sotnikov-s — COMMENTED** [review]
the rest LGTM

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/10947680518

**Review by quasisamurai — DISMISSED** [review]

**Review by sotnikov-s — DISMISSED** [review]

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/11022410836

**Review by NicholasDotSol — APPROVED** [review]
really good change

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/11040084760

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**